### PR TITLE
Extend env variables to fix literal string parsing

### PR DESCRIPTION
**Pull request by Aishwarya-Lad**
Please read [CONTRIBUTING.md](CONTRIBUTING.md) for details on our [code of conduct](CODE_OF_CONDUCT.md), and the process for submitting pull requests to us.

Fixes #

## Proposed Changes

-
-
-

### Description
Env variables are not getting parsed correctly when passed on to the cache plugin. 

Extended the env variables before setting the cache directory
### Checklist

- [x] Read the [**CONTRIBUTING**](CONTRIBUTING.md) document.
- [x] Read the [**CODE OF CONDUCT**](CODE_OF_CONDUCT.md) document.
- [ ] Add tests to cover changes.
- [x] Ensure your code follows the code style of this project.
- [x] Ensure CI and all other PR checks are green OR
    - [x] Code compiles correctly.
    - [ ] Created tests which fail without the change (if possible).
    - [ ] All new and existing tests passed.
- [ ] Add your changes to `Unreleased` section of [CHANGELOG](CHANGELOG.md).
- [ ] Improve and update the [README](README.md) (if necessary).
- [ ] Ensure [documentation](./DOCS.md) is up-to-date. The same file will be updated in [plugin index](https://github.com/drone/drone-plugin-index/blob/master/content/meltwater/drone-cache/index.md) when your PR is accepted, so it will be available for end-users at http://plugins.drone.io.

### CODE REVIEW

File: archive/tar/tar.go [hunk 1]
```diff
@@ -48,6 +48,7 @@ func (a *Archive) Create(srcs []string, w io.Writer, isRelativePath bool) (int64
 	var written int64
 
 	for _, src := range srcs {
+	    src = os.ExpandEnv(src)
```
Comment (jamie-harness) [inline archive/tar/tar.go:51]: change this in restorer.go instead of here. We should have it in upstream already

**Review by jamie-harness — CHANGES_REQUESTED** [review]

File: archive/tar/tar.go (hunk 1 above)
Comment (Aishwarya-Lad) [reply archive/tar/tar.go]: Done

File: cache/rebuilder.go [hunk 2]
```diff
@@ -55,6 +55,10 @@ func (r rebuilder) Rebuild(srcs []string) error {
 	)
 
 	for _, src := range srcs {
+		level.Info(r.logger).Log("msg", "Before expansion", "src", src)
+		src = os.ExpandEnv(src)
+		level.Info(r.logger).Log("msg", "After expansion", "src", src)
+		level.Info(r.logger).Log("msg", "os.ExpandEnv($HOME)", "test", os.ExpandEnv("$HOME"))
```
Comment (hemanthmantri) [inline cache/rebuilder.go:61]: These should be debug level IMHO. Alternatively, we can consider printing one log line as follows:
"Source %s is resolved to %s"

File: cache/rebuilder.go [hunk 3]
```diff
@@ -76,9 +80,6 @@ func (r rebuilder) Rebuild(srcs []string) error {
 			}
 		}
 
-		level.Info(r.logger).Log("msg", "rebuilding cache for directory", "local", src)
-		level.Debug(r.logger).Log("msg", "rebuilding cache for directory", "remote", dst)
-
```
Comment (hemanthmantri) [inline cache/rebuilder.go:81]: Do we know why are we removing this?

File: cache/restorer.go [hunk 4]
```diff
@@ -73,7 +74,9 @@ func (r restorer) Restore(dsts []string) error {
 
 	for _, dst := range dsts {
 		src := filepath.Join(namespace, key, dst)
-
+        level.Info(r.logger).Log("srcs", "Before expansion ", src)
+        src = os.ExpandEnv(src)
+        level.Info(r.logger).Log("srcs", "After expansion ", src)
```
Comment (hemanthmantri) [inline cache/restorer.go:79]: Consider just one log line as mentioned above.

**Pull request closed without merging**